tsfc/fem.py

```diff
@@ -108,6 +112,88 @@ def translator(self):
         return Translator(self)
 
 
+class CoordinateMapping(PhysicalGeometry):
+    """Callback class that provides physical geometry to FInAT elements.
+
+    Required for elements whose basis transformation requires physical
+    geometry such as Argyris and Hermite.
+
+    :arg mt: The modified terminal whose element will be tabulated.
+    :arg interface: The interface carrying information (generally a
+        :class:`PointSetContext`).
+    """
+    def __init__(self, mt, interface):
+        super().__init__()
+        self.mt = mt
+        self.interface = interface
+
+    @property
+    def config(self):
+        config = {name: getattr(self.interface, name)
+                  for name in ["ufl_cell", "precision", "index_cache"]}
+        config["interface"] = self.interface
+        return config
+
+    def cell_size(self):
+        return self.interface.cell_size(self.mt.restriction)
+
+    def jacobian_at(self, point):
+        expr = Jacobian(self.mt.terminal.ufl_domain())
+        if self.mt.restriction == '+':
+            expr = PositiveRestricted(expr)
+        elif self.mt.restriction == '-':
+            expr = NegativeRestricted(expr)
+        expr = preprocess_expression(expr)
+
+        config = {"point_set": PointSingleton(point)}
+        config.update(self.config)
+        context = PointSetContext(**config)
+        return map_expr_dag(context.translator, expr)
+
+    def reference_normals(self):
+        return gem.Literal(numpy.asarray([self.interface.fiat_cell.compute_normal(i) for i in range(3)]))
```

Comment:
Yes, FIAT `compute_normal` gives normals pointing either in or out, consistent with the Rognes et al. (2009) numbering.